woocommerce/wc-admin
# Remote Inbox: Add feature to list of optional feature flags

Add Remote Inbox Notifications to the list of optional feature flags. This exposes the feature to the repurposed `woocommerce_admin_disabled` filter. Since a merchant facing toggle already exists behind the `woocommerce_show_marketplace_suggestions` option, none was added to WooCommerce > Settings > Advanced > Features.

See pcShBQ-1i#comment-187 for more discussion.

### Detailed test instructions:

1. Ensure Marketplace suggestions are on, `wp option set woocommerce_show_marketplace_suggestions yes`.
2. Turn off all optional features by placing `add_filter( 'woocommerce_admin_disabled', '__return_true' );` somewhere or in another plugin.
4. Remove all remote notes from your inbox.
```
DELETE FROM `wp_wc_admin_notes` WHERE source = 'woocommerce.com'
```
5. Trigger the `wc_admin_daily` using Cron Events plugin
6. Query the database and ensure no remote inbox notes were added. 
```
SELECT * FROM `wp_wc_admin_notes` WHERE source = 'woocommerce.com'
```
7. Now, remove the filter `add_filter( 'woocommerce_admin_disabled', '__return_true' );` to turn on optional features.
8. Again, trigger `wc_admin_daily` event and query the database for inbox notifications coming from woocommerce.com. There should now be plenty.

### Bonus test

Confirm other features that use the Remote Inbox Notification rule system still function.

1. Turn off all optional features by placing `add_filter( 'woocommerce_admin_disabled', '__return_true' );` somewhere or in another plugin.
2. Remove any cached results from the free extensions list feature, `wp transient delete woocommerce_admin_remote_free_extensions_specs`.
2. Visit `/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=business-features` in the OBW
3. Select Free features
4. Confirm a Network request to `/free-extensions` was made and see the list populated in the UI without error.
